README.md

```diff
@@ -0,0 +1,580 @@
+requirements.txt` to constrain the dev requirements to packages already
+selected for production in `requirements.txt`.
+
+```
+# dev-requirements.in
+-c requirements.txt
+django-debug-toolbar<2.2
+```
+
+First, compile `requirements.txt` as usual:
+
+```
+$ pip-compile
+#
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
+#
+#    pip-compile
+#
+django==2.1.15
+    # via -r requirements.in
+pytz==2023.3
+    # via django
+```
+
+Now compile the dev requirements and the `requirements.txt` file is used as
+a constraint:
+
+```console
+$ pip-compile dev-requirements.in
+#
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
+#
+#    pip-compile dev-requirements.in
+#
+django==2.1.15
+    # via
+    #   -c requirements.txt
+    #   django-debug-toolbar
+django-debug-toolbar==2.1
+    # via -r dev-requirements.in
+pytz==2023.3
+    # via
+    #   -c requirements.txt
+    #   django
+sqlparse==0.4.3
+    # via django-debug-toolbar
+```
+
+As you can see above, even though a `2.2` release of Django is available, the
+dev requirements only include a `2.1` version of Django because they were
+constrained. Now both compiled requirements files can be installed safely in
+the dev environment.
+
+To install requirements in production stage use:
+
+```console
+$ pip-sync
+```
+
+You can install requirements in development stage by:
+
+```console
+$ pip-sync requirements.txt dev-requirements.txt
+```
+
+### Version control integration
+
+You might use `pip-compile` as a hook for the [pre-commit](https://github.com/pre-commit/pre-commit).
+See [pre-commit docs](https://pre-commit.com/) for instructions.
+Sample `.pre-commit-config.yaml`:
+
+```yaml
+repos:
+    - repo: https://github.com/jazzband/pip-tools
+    rev: 6.13.0
+    hooks:
+        - id: pip-compile
+```
+
+You might want to customize `pip-compile` args by configuring `args` and/or `files`, for example:
+
+```yaml
+repos:
+    - repo: https://github.com/jazzband/pip-tools
+    rev: 6.13.0
+    hooks:
+        - id: pip-compile
+        files: ^requirements/production\.(in|txt)$
+        args: [--index-url=https://example.com, requirements/production.in]
+```
+
+If you have multiple requirement files make sure you create a hook for each file.
+
+```yaml
+repos:
+    - repo: https://github.com/jazzband/pip-tools
+    rev: 6.13.0
+    hooks:
+        - id: pip-compile
+        name: pip-compile setup.py
+        files: ^(setup\.py|requirements\.txt)$
+        - id: pip-compile
+        name: pip-compile requirements-dev.in
+        args: [requirements-dev.in]
+        files: ^requirements-dev\.(in|txt)$
+        - id: pip-compile
+        name: pip-compile requirements-lint.in
+        args: [requirements-lint.in]
+        files: ^requirements-lint\.(in|txt)$
+        - id: pip-compile
+        name: pip-compile requirements.txt
+        args: [requirements.txt]
+        files: ^requirements\.(in|txt)$
```

Comment:
Make sure to incorporate https://github.com/jazzband/pip-tools/pull/1847 and other commits that changed `README.rst` post #1862: https://github.com/jazzband/pip-tools/commits/main/README.rst.